src/main/java/me/zeroeightsix/kami/mixin/client/MixinClientPlayNetworkHandler.java

```diff
@@ -0,0 +1,51 @@
+package me.zeroeightsix.kami.mixin.client;
+
+import me.zeroeightsix.kami.feature.module.SignSpammer;
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.block.entity.SignBlockEntity;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayNetworkHandler;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.network.Packet;
+import net.minecraft.network.packet.c2s.play.UpdateSignC2SPacket;
+import net.minecraft.network.packet.s2c.play.SignEditorOpenS2CPacket;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+@Mixin(ClientPlayNetworkHandler.class)
+public abstract class MixinClientPlayNetworkHandler {
+    @Shadow
+    public abstract void sendPacket(Packet<?> packet);
+
+    @Inject(
+            method = "onSignEditorOpen",
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/client/network/ClientPlayerEntity;openEditSignScreen(Lnet/minecraft/block/entity/SignBlockEntity;)V"
+            ),
+            cancellable = true,
+            locals = LocalCapture.CAPTURE_FAILSOFT
+    )
+    private void onSignEditorOpen(SignEditorOpenS2CPacket packet, CallbackInfo ci, BlockEntity blockEntity) {
+        // this variable is just here for some deduplication
+        SignSpammer spammer = SignSpammer.INSTANCE;
```

Comment:
I don't understand what this is supposed to mean. Do you mean avoiding typing SignSpammer.INSTANCE a bunch of times? If so, I think it doesn't matter if the Mixins aren't as pretty due to being in Java

Reply:
Yes, that's exactly what it means. I think it's still nice to have tho and makes the code look cleaner. Let's see what 086 thinks of it.

Reply:
> I think it doesn't matter if the Mixins aren't as pretty due to being in Java
Java can still be pretty. I'm happy with this variable.